src/include/storage/storage_defs.h

```diff
@@ -15,19 +15,42 @@ namespace terrier::storage {
 // BlockLayout as a runtime object, instead baking them in as compiled code
 // (Think of this as writing the class with a BlockLayout template arg, except
 // template instantiation is done by LLVM at runtime and not at compile time.
+/**
+ * Stores metadata about the layout of a block.
+ * This will eventually be baked in as compiled code by LLVM.
+ */
 struct BlockLayout {
+  /**
+   * Constructs a new block layout.
+   * @param num_attrs number of attributes.
+   * @param attr_sizes vector of attribute sizes.
+   */
   BlockLayout(const uint16_t num_attrs, std::vector<uint8_t> attr_sizes)
       : num_cols_(num_attrs),
         attr_sizes_(std::move(attr_sizes)),
         tuple_size_(ComputeTupleSize()),
         header_size_(HeaderSize()),
         num_slots_(NumSlots()) {}
 
+  /**
+   * Number of columns.
+   */
   const uint16_t num_cols_;
+  /**
+   * Vector of attribute sizes.
+   */
   const std::vector<uint8_t> attr_sizes_;
-  // Cached so we don't have to iterate through attr_sizes every time
+  /**
+   * Cached tuple size so that we don't have to iterate through attr_sizes_ every time.
```

Comment:
again, cached or not is an implementation detail better left for inline comments. (intended for developer in this file rather than users of the file.)